tests/explainers/test_generic_explainer.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from pyreal.explainers import Explainer
+from pyreal.transformers import FeatureSelectTransformer
+
+
+def test_init_invalid_transforms(regression_no_transforms):
+    invalid_transform = "invalid"
+    with pytest.raises(TypeError):
+        Explainer(regression_no_transforms["model"], regression_no_transforms["x"],
+                  m_transformers=invalid_transform)
+    with pytest.raises(TypeError):
+        Explainer(regression_no_transforms["model"], regression_no_transforms["x"],
+                  e_transformers=invalid_transform)
+    with pytest.raises(TypeError):
+        Explainer(regression_no_transforms["model"], regression_no_transforms["x"],
+                  i_transformers=invalid_transform)
+
+
+def test_init_invalid_model():
+    invalid_model = []
+    with pytest.raises(TypeError):
+        Explainer(invalid_model, pd.DataFrame([0]))
+
+
+def test_run_transformers(regression_one_hot):
+    x = pd.DataFrame([[2, 1, 3],
+                      [4, 3, 4],
+                      [6, 7, 2]], columns=["A", "B", "C"])
+    expected = pd.DataFrame([[1, 3, 1, 0, 0],
+                             [3, 4, 0, 1, 0],
+                             [7, 2, 0, 0, 1]], columns=["B", "C", "A_2", "A_4", "A_6"])
+
+    feature_select = FeatureSelectTransformer(columns=["B", "A_2"])
+    explainer = Explainer(regression_one_hot["model"], x,
+                          e_transformers=regression_one_hot["transformers"],
+                          m_transformers=feature_select,
+                          i_transformers=regression_one_hot["transformers"])
+    result = explainer.transform_to_x_interpret(x)
+    assert_frame_equal(result, expected, check_like=True, check_dtype=False)
+    result = explainer.transform_to_x_model(x)
+    assert_frame_equal(result, expected[["B", "A_2"]], check_like=True, check_dtype=False)
+    result = explainer.transform_to_x_explain(x)
+    assert_frame_equal(result, expected, check_like=True, check_dtype=False)
+
+
+def test_predict_regression(regression_no_transforms, regression_one_hot):
```

Comment:
So for this and all following unit tests, you are assuming `Explainer` uses a ShapExplainer under the hood. This is true for now but doesn't necessarily have to be in the future, which would cause this test to break.

The easiest way to address this would be to add an "algorithm" keyword to the Explainer class that allows the user to specific the algorithm (which would include all the options from all other Explainers. We can automate the process of getting this list in a future PR, for now you can manually add it). Then, use the Shap keyword for these tests

Reply:
I agree that the default `Explainer` object should be able to use different algorithms in the future, but I think the users are not responsible for selecting the **algorithm**. They could just use specific explainers in that case. 
My solution to this would be adding an `algorithm` attribute but not a keyword to the `Explainer` class.

Reply:
The algorithm would be optional, and users would not generally use it - this matches the format of other explainers. The benefit of doing it this way is it allows users to easily switch out the algorithms and try new things, without needing to change to a new explainer type. If the user does not provide an algorithm keyword, then we select for them